src/main/java/io/confluent/connect/elasticsearch/DataConverter.java

```diff
@@ -52,9 +54,11 @@
   }
 
   private final boolean useCompactMapEntries;
+  private final BehaviorOnNullValues behaviorOnNullValues;
 
-  public DataConverter(boolean useCompactMapEntries) {
+  public DataConverter(boolean useCompactMapEntries, BehaviorOnNullValues behaviorOnNullValues) {
     this.useCompactMapEntries = useCompactMapEntries;
+    this.behaviorOnNullValues = behaviorOnNullValues;
```

Comment:
Can `behaviorOnNullValues` parameter be null? Seems like we don't really use asserts (which I like very much for stating intent), but alternatively here you could just assume the default value for safety.

Reply:
As of right now, none of the instantiations of the `DataConverter` within the project are at risk of passing in `null` for the `behaviorOnNullValues` parameter. I'm hesitant to allow `null` as a value here and just implicitly replace it with `DEFAULT`; seems better to, as you pointed out, utilize an assert to ensure it isn't `null`. At the very least, if `DEFAULT` is used as a replacement for `null`, I think a Javadoc comment should be added above the class detailing this behavior just to make it easier should somebody else try to use the class. Alternatively, we could overload the constructor with a (well-documented) one-arg variant that then calls the two-arg constructor with `DEFAULT` as the second argument, which we can then safely assert as non-null.

Mostly, just don't like the idea of adding implicit behavior and want to avoid any potential future confusion to arise from that.

Reply:
Let's not overload the constructor if we don't need it. If the goal is that the supplied parameter never is null, then let's add an assertion and JavaDoc that explains this.

Reply:
Will do!